main_test.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockDownloader struct {
```

Comment:
I know we hand-rolled all this at the start, but maybe we should move to using `github.com/stretchr/testify/mock` now. It's being used on search-service and we already use the `assert` package. Although I like that you don't need to learn another package to figure out what is going on, the manualness can be a bit tiring.

Reply:
Done, turns out that's a really nice library.